### PR TITLE
refactor: drop Node 6/8 support & upgrade dependencies

### PR DESCRIPTION
**Pull request by aaarichter**
### Breaking Change
- [x] drop Node 6/8 support

### Changes
- [x] chore: upgrade dependencies
- [x] style: apply JS transforms to `lib` and `test`

### CODE REVIEW

**Comment by dbushong** [comment]
As long as we're major-bumping to drop Node 6 compat anyway, why not also drop Node 8 compat since it's no longer officially supported?

**Comment by aaarichter** [comment]
> As long as we're major-bumping to drop Node 6 compat anyway, why not also drop Node 8 compat since it's no longer officially supported?

sounds good

**Comment by aaarichter** [comment]
@dbushong further changes:
- dropped Node 8 support
- applied JS transform to lib and test, making `co` library unnecessary

File: lib/github/parse-repository.js [hunk 1]
```diff
@@ -44,18 +42,19 @@ function parseRepository(repositoryField) {
 
   if (!repository || typeof repository !== 'string') {
     throw new Error(`Invalid or missing repository field: ${repositoryField}`);
-  }
+  } // Transform git@ into git+ssh://git@ which URL constructor can handle
 
-  // Transform git@ into git+ssh://git@ which Url.parse can handle
```
Comment (dbushong) [inline lib/github/parse-repository.js]: cleanup, pls

File: lib/github/client.js [hunk 2]
```diff
@@ -41,24 +41,27 @@ const parseRepository = require('./parse-repository');
 function Github(config) {
   Gofer.call(this, config, 'github');
 }
-util.inherits(Github, Gofer);
 
+util.inherits(Github, Gofer);
 Github.prototype.registerEndpoints({
```
Comment (dbushong) [inline lib/github/client.js]: maybe a quick refactor to make this idiomatic for modern gofer?
```js
class Github extends Gofer {
  constructor(config) {
    super(config, 'github', require('../../package.json').version);
  }
  pull() {
    return this.get('/repos/{owner}/{repo}/pulls/{pullId}/commits', {
      pathParams: { pullId },
      endpointName: 'pull',
    }).json();
  }
```

etc etc  (...or save it for another PR :-) )

File: bin/nlm.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
 #!/usr/bin/env node
+
 'use strict';
+
```
Comment (dbushong) [inline bin/nlm.js]: for files like this i'd prefer we just `git reset` them and skip the commit noise; in general i'd go ahead and remove any post-automated-transform comment or whitespace changes that aren't needed

File: lib/commands/changelog.js [hunk 4]
```diff
@@ -53,8 +53,9 @@ const showChangelog = co.wrap(function*(cwd, pkg, options) {
 
   const tasks = [ensureVersionTag, getPendingChanges, generateChangeLog];
 
-  for (const task of tasks) yield Promise.resolve(runTask(task));
+  for (const task of tasks) await Promise.resolve(runTask(task));
```
Comment (dbushong) [inline lib/commands/changelog.js]:
```suggestion
  for (const task of tasks) await runTask(task);
```

`yield` bombed if it wasn't handed a Promise, but `await` doesn't mind.

File: lib/commands/release.js [hunk 5]
```diff
@@ -81,31 +88,35 @@ function release(cwd, pkg, options) {
...
       return null;
     }
+
     if (options.releaseType === 'none') {
       debug('No changes; just verifying NPM publish');
       publishTasks = [publishToNpm];
     }
 
-    for (const task of publishTasks) yield Promise.resolve(runTask(task));
+    for (const task of publishTasks) await Promise.resolve(runTask(task));
```
Comment (dbushong) [inline lib/commands/release.js]: ditto

File: lib/commands/verify.js [hunk 6]
```diff
@@ -72,10 +79,10 @@ function verify(cwd, pkg, options) {
     if (err.code === 'ENOENT') {
       return Promise.resolve();
     }
+
     throw err;
-  }
+  } // Not making this configurable to prevent some possible abuse
 
-  // Not making this configurable to prevent some possible abuse
```
Comment (dbushong) [inline lib/commands/verify.js]: cleanup

File: lib/commands/verify.js [hunk 7]
```diff
@@ -105,10 +112,11 @@ function verify(cwd, pkg, options) {
     return task(cwd, pkg, options);
   }
 
-  const runVerifyTasks = co.wrap(function*() {
-    for (const task of verifyTasks) yield Promise.resolve(runTask(task));
-  });
+  async function runVerifyTasks() {
+    for (const task of verifyTasks) await Promise.resolve(runTask(task));
```
Comment (dbushong) [inline lib/commands/verify.js]: ditto

File: lib/git/ensure-tag.js [hunk 8]
```diff
@@ -33,14 +33,16 @@
...
+
 const fs = require('fs');
 
 const run = require('../run');
 
 function fetchTag(cwd, tag) {
-  return run('git', ['fetch', 'origin', 'tag', tag], { cwd: cwd });
+  return run('git', ['fetch', 'origin', 'tag', tag], {
+    cwd,
+  });
```
Comment (dbushong) [inline lib/git/ensure-tag.js]: babel verticalizes everything - if you see stuff like this, please just undo it

File: lib/github/parse-repository.js [hunk 9]
```diff
@@ -44,18 +42,19 @@ function parseRepository(repositoryField) {
...
   const url = repository.replace(/^git@([^:]+):(.*)$/, 'git+ssh://git@$1/$2');
-  const parsed = Url.parse(url);
+  const parsed = new URL(url, 'git+ssh://git@github.com');
```
Comment (dbushong) [inline lib/github/parse-repository.js]: are we better off leaving off the base url 2nd arg and letting it die if it's some weird relative url?

Comment (aaarichter) [reply lib/github/parse-repository.js]: `new URL` fails on these shorthand git urls that are covered by the tests

File: lib/github/client.js (hunk 2 above)
Comment (aaarichter) [reply lib/github/client.js]: yes a follow up

File: lib/github/parse-repository.js (hunk 9 above)
Comment (dbushong) [reply lib/github/parse-repository.js]: oh, people putting like `url: 'myusername/foo'`?  yuck.  but yeah, good call.

File: lib/steps/version-commit.js [hunk 10]
```diff
@@ -41,38 +42,42 @@ const NLM_GIT_NAME = 'nlm';
...
+  if (!fs.existsSync(lockPath)) return; // if package-lock.json is in the .gitignore, then we just generated it
   // and shouldn't commit it
+
   const ignorePath = path.join(cwd, '.gitignore');
+
   if (fs.existsSync(ignorePath)) {
     const ignore = fs.readFileSync(ignorePath, 'utf8');
     if (/^\/?package-lock\.json\s*$/m.test(ignore)) return;
-  }
+  } // we don't want to risk re-sorting the file, so just do a regexp replace
```
Comment (dbushong) [inline lib/steps/version-commit.js]: cleanup, pls

**Pull request merged**